File: python_solutions/bee1118_varias_notas_com_validacao.py
# Walber Araújo 16/11/2023

def calcular_media():
    qtd_validas = 0
    soma = 0
    while qtd_validas < 2:
        nota = float(input())
        if 0 <= nota <= 10:
            qtd_validas += 1
            soma += nota
        else: print('nota invalida')

    return soma / 2

while True:
    media = calcular_media()
    print(f'media = {media:.2f}')
    while True:
        continuar = input('novo calculo (1-sim 2-nao)\n')
        if continuar == '1' or continuar == '2': break
    if continuar == '2': break

File: java_solutions/bee1118_varias_notas_com_validacao.java
//Walber Araújo 17/11/2023

import java.util.Locale;
import java.util.Scanner;

public class bee1118_varias_notas_com_validacao {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        int confirma;

        do {
            byte qtd_Notas_Validas = 0;
            float soma = 0;

            while (qtd_Notas_Validas < 2) {
                float nota = sc.nextFloat();
                if (nota >= 0.0 && nota <= 10.0) {
                    qtd_Notas_Validas++;
                    soma += nota;
                }
                else {
                    System.out.println("nota invalida");
                }
            }
            System.out.printf("media = %.2f%n", soma / 2.0);
            do {
                System.out.println("novo calculo (1-sim 2-nao)");
                confirma = sc.nextInt();
            } while (confirma != 1 && confirma != 2);

        } while(confirma == 1);
    }
}